tests/integration/test_get_deposits.py

```diff
@@ -423,3 +432,212 @@ def test_get_deposit_with_permissions_json_serializer_returns_serialized_permiss
+    webhook.subscribers.append(subscriber)
+
+    db.session.add(repo)
+    db.session.add(webhook)
+    db.session.commit()
+
+    # get the deposit
+    #################
+    pid = deposit['_deposit']['id']
+    metadata = deposit.get_record_metadata()
+    bucket = deposit.files.bucket
+    headers = auth_headers_for_example_user + [('Accept', 'application/form+json')]
+
+    resp = client.get(f'/deposits/{pid}', headers=headers)
+
+    assert resp.status_code == 200
+    assert resp.json == {
+        'access': {
+            'deposit-admin': {'roles': [], 'users': [example_user.email]},
+            'deposit-read': {'roles': [], 'users': [example_user.email]},
+            'deposit-update': {'roles': [], 'users': [example_user.email]}
+        },
+        'created_by': example_user.email,
+        'is_owner': True,
+        'can_admin': True,
+        'can_update': True,
+        'experiment': 'CMS',
+        'created': metadata.created.strftime('%Y-%m-%dT%H:%M:%S.%f+00:00'),
+        'updated': metadata.updated.strftime('%Y-%m-%dT%H:%M:%S.%f+00:00'),
+        'id': pid,
+        'links': {
+            'bucket': f'http://analysispreservation.cern.ch/api/files/{str(bucket)}',
+            'clone': f'http://analysispreservation.cern.ch/api/deposits/{pid}/actions/clone',
+            'discard': f'http://analysispreservation.cern.ch/api/deposits/{pid}/actions/discard',
+            'edit': f'http://analysispreservation.cern.ch/api/deposits/{pid}/actions/edit',
+            'files': f'http://analysispreservation.cern.ch/api/deposits/{pid}/files',
+            'html': f'http://analysispreservation.cern.ch/drafts/{pid}',
+            'permissions': f'http://analysispreservation.cern.ch/api/deposits/{pid}/actions/permissions',
+            'publish': f'http://analysispreservation.cern.ch/api/deposits/{pid}/actions/publish',
+            'self': f'http://analysispreservation.cern.ch/api/deposits/{pid}',
+            'upload': f'http://analysispreservation.cern.ch/api/deposits/{pid}/actions/upload'
+        },
+        'metadata': {}, 'revision': 1, 'status': 'draft', 'type': 'deposit',
+        'labels': [], 'workflows': [],
+        'schema': {'name': 'test-schema', 'version': '1.0.0'},
+        'schemas': {
+            'schema': {
+                'title': 'deposit-test-schema',
+                'type': 'object',
+                'properties': {
+                    'date': {'type': 'string'},
+                    'title': {'type': 'string'}},
+                },
+            'uiSchema': {
+                'title': 'ui-test-schema',
+                'type': 'object',
+                'properties': {
+                    'field': {'type': 'string'},
+                    'title': {'type': 'string'}},
+                }
+        },
+        'files': [],
+        'webhooks': [{
+            'branch': 'mybranch',
+            'event_type': 'push',
+            'snapshots': []
+        }]
+    }
```

Comment:
In general your test was ok, just forgot to check few fields, could be sth like this:
```
def test_get_deposit_when_form_json_serializer_returns_correct_fields(client, db,
                                                 auth_headers_for_example_user,
                                                 example_user, create_deposit,
                                                 create_schema, file_tar,
                                                 github_release_webhook):  # create github_release_webhook fixture in a conftest

    create_schema('test-schema',
                  experiment='CMS',
                  deposit_schema={'title': 'deposit-test-schema'},
                  deposit_options={'title': 'ui-test-schema'})
    deposit = create_deposit(example_user,
                             'test-schema', 
                             { # you forgot to test metadata
                                 '$ana_type': 'test-schema',
                                 'my_field': 'mydata'
                             },
                             experiment='CMS',
                             files={'readme': BytesIO(b'Hello!')})  # thats how you can test file uploaded with analysis
    pid = deposit['_deposit']['id']
    metadata = deposit.get_record_metadata()
    file = deposit.files['readme']
    subscriber = GitWebhookSubscriber(record_id=deposit.id,
                                      user_id=example_user.id)
    snapshot = GitSnapshot(  # you forgot to test snapshots, we need to add some
        payload={
            'branch': None,
            'event_type': 'release',
            'commit': None,
            'author': {
                'name': 'owner',
                'id': 1
            },
            'link': 'https://github.com/owner/test/releases/tag/v1.0.0',
            'release': {
                'tag': 'v1.0.0',
                'name': 'test release 1'
            }
        })
    github_release_webhook.snapshots.append(snapshot)
    github_release_webhook.subscribers.append(subscriber)
    db.session.commit()

    resp = client.get(f'/deposits/{pid}',
                      headers=auth_headers_for_example_user +
                      [('Accept', 'application/form+json')])

    assert resp.status_code == 200
    assert resp.json == {
        'access': {
            'deposit-admin': {
                'roles': [],
                'users': [example_user.email]
            },
            'deposit-read': {
                'roles': [],
                'users': [example_user.email]
            },
            'deposit-update': {
                'roles': [],
                'users': [example_user.email]
            }
        },
        'created_by': example_user.email,
        'is_owner': True,
        'can_admin': True,
        'can_update': True,
        'experiment': 'CMS',
        'created': metadata.created.strftime('%Y-%m-%dT%H:%M:%S.%f+00:00'),
        'updated': metadata.updated.strftime('%Y-%m-%dT%H:%M:%S.%f+00:00'),
        'id': pid,
        'links': {
            'bucket': f'http://analysispreservation.cern.ch/api/files/{str(file.bucket)}',
            'clone': f'http://analysispreservation.cern.ch/api/deposits/{pid}/actions/clone',
            'discard': f'http://analysispreservation.cern.ch/api/deposits/{pid}/actions/discard',
            'edit': f'http://analysispreservation.cern.ch/api/deposits/{pid}/actions/edit',
            'files': f'http://analysispreservation.cern.ch/api/deposits/{pid}/files',
            'html': f'http://analysispreservation.cern.ch/drafts/{pid}',
            'permissions': f'http://analysispreservation.cern.ch/api/deposits/{pid}/actions/permissions',
            'publish': f'http://analysispreservation.cern.ch/api/deposits/{pid}/actions/publish',
            'self': f'http://analysispreservation.cern.ch/api/deposits/{pid}',
            'upload': f'http://analysispreservation.cern.ch/api/deposits/{pid}/actions/upload'
        },
        'metadata': {
            'my_field': 'mydata'
        },
        'revision': 2,
        'status': 'draft',
        'type': 'deposit',
        'labels': [],
        'schema': {
            'name': 'test-schema',
            'version': '1.0.0'
        },
        'schemas': {
            'schema': {
                'title': 'deposit-test-schema',
            },
            'uiSchema': {
                'title': 'ui-test-schema',
            }
        },
        'files': [{
            'bucket': str(file.bucket),
            'checksum': file.file.checksum,
            'key': file.key,
            'size': file.file.size,
            'version_id': str(file.version_id)
        }],
        'webhooks': [{
            'branch': 'mybranch',
            'event_type': 'release',
            'host': 'github.com',
            'name': 'repository',
            'owner': 'owner',
            'snapshots': [{
                'created': snapshot.created.strftime(
                    '%Y-%m-%dT%H:%M:%S.%f+00:00'),
                'payload': {
                    'author': {
                        'id': 1,
                        'name': 'owner'
                    },
                    'branch': None,
                    'commit': None,
                    'event_type': 'release',
                    'link': 'https://github.com/owner/test/releases/tag/v1.0.0',
                    'release': {
                        'name': 'test release 1',
                        'tag': 'v1.0.0'
                    }
                }
            }]
        }]
    }